pkg/utils/flow/taskfn.go

```diff
@@ -53,6 +53,12 @@ func (t TaskFn) DoIf(condition bool) TaskFn {
 	return t.SkipIf(!condition)
 }
 
+// DoIfFn returns a TaskFn that will be executed if the invocation of conditionalFn is true when it is called.
+// Otherwise, it will do nothing when called.
+func (t TaskFn) DoIfFn(conditionalFn func() bool) TaskFn {
+	return t.SkipIf(!conditionalFn())
```

Comment:
Why do not `t.DoIf(conditionalFn())`? 😕

Reply:
I wanted to construct the graph and compile it to a Flow without evaluating predicates during construction phase. And re-use the compiled flow repeatedly using different closure states when its run. Let me test this again just to confirm that this is then possible using existing `DoIf`

Reply:
```go
		It("should process the function only when the conditional function returns true", func() {
			results := make([]string, 0, 2)
			var trueFnEvaluated bool
			var falseFnEvaluated bool
			trueFn := func() bool {
				trueFnEvaluated = true
				return true
			}
			falseFn := func() bool {
				falseFnEvaluated = true
				return false
			}
			var (
				g = flow.NewGraph("foo")
				_ = g.Add(flow.Task{Name: "a-task", Fn: flow.TaskFn(func(ctx context.Context) error {
					results = append(results, "a-task-result")
					return nil
				}).DoIf(trueFn())})
				_ = g.Add(flow.Task{Name: "b-task", Fn: flow.TaskFn(func(ctx context.Context) error {
					results = append(results, "b-task-result")
					return nil
				}).DoIf(falseFn())})
				f = g.Compile()
			)
			Expect(trueFnEvaluated).To(BeFalse())
			Expect(falseFnEvaluated).To(BeFalse())
			Expect(f.Run(ctx, flow.Opts{})).ToNot(HaveOccurred())
			Expect(results).To(HaveLen(1))
			Expect(results[0]).To(Equal("a-task-result"))
		})

```
In the test above you can see that the predicate functions are actually evaluated at construction time and before flow.Run is called. This is not what was desired.

Reply:
I cannot follow. My suggestions was only to change 

```suggestion
	return t.DoIf(conditionalFn())
```

to make it easier readable. 

Also, can we introduce a similar `SkipIfFn` for symmetry reasons?

Also, please check https://github.com/gardener/gardener/pull/5801#issuecomment-1097725099

Reply:
Ok the only way to have a lazy evaluation of a predicate is to have the predicate as part of the TaskFn itself. I will abandon this PR as the proposed change anyways does not solve what i originally intended. Should have done better testing/validation of the intent. My bad.